backend/agents/grading_agent.py

```
@@ -90,16 +95,33 @@ def calculate_overall_score(scene_scores: str) -> str:
+                
+                # Parse weights if provided
+                weight_list = []
+                if weights:
+                    for w in weights.split(','):
+                        w = w.strip()
+                        if w.replace('.', '').isdigit():
+                            weight_list.append(float(w))
+                
+                if scores:
+                    if weight_list and len(weight_list) == len(scores):
+                        weighted_score = sum(s * w for s, w in zip(scores, weight_list))
```

Ruff (0.14.0)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)